src/socket.c

```diff
@@ -3116,45 +3206,116 @@ static z_stream *new_zstream(void)
 }
 #endif /* HAVE_MCCP */
 
-/* handle input from current socket */
-static int handle_socket_input(const char *simbuffer, int simlen)
+#if WIDECHAR
+static int inbound_decode_str(String *output, String *input, UConverter *conv, const char cflush)
 {
-    char rawchar, localchar, inbuffer[4096];
-    const char *buffer, *place;
+    int shiftby = 0;
+    if (input->data < input->data + input->len) {
+        shiftby = inbound_decode(output, input->data, input->data + input->len, conv, cflush);
+    }
+    if (shiftby > 0) {
+        if (cflush == 0) {
+            Stringshift(input, shiftby);
+        } else {
+            Stringtrunc(input, 0);
+        }
+    }
+    return shiftby;
+}
+/* Take input of a particular encoding and Stringcat it into 'output'.
+ * Flush shold be FALSE(0) unless the socket is being shut down.
+ * Returns number of bytes the input pointer should increment.
+ * Only modifies 'output' and 'conv' arguments.
+ */
+static int inbound_decode(String *output, const char *input, const char *iendptr, UConverter *conv, const char cflush)
+{
+    /* This definitely, definitely needs optimizations for
+          * ISO-8859-1 -> UTF-8, and UTF-8 -> UTF-8. */
+    const char *iptr = input;
+
+    UChar outbufferUTF16[BUFFSIZE*4];
+    UChar *optr = outbufferUTF16;
+    const UChar *oendptr = outbufferUTF16 + BUFFSIZE*4;
+    UErrorCode err16 = U_ZERO_ERROR;
+
+    char outbufferUTF8[BUFFSIZE*8];
+    UErrorCode err8 = U_ZERO_ERROR;
+    int32_t utf8written = 0;
+
+    UBool flush = cflush ? TRUE : FALSE;
+    
+    /* xcharset -> UTF-16 -> UTF-8 */
+    ucnv_toUnicode(conv, &optr, oendptr, &iptr, iendptr, NULL, flush, &err16);
+/*
+void ucnv_toUnicode 	( 	UConverter *  	converter,
+		UChar **  	target,
+		const UChar *  	targetLimit,
+		const char **  	source,
+		const char *  	sourceLimit,
+		int32_t *  	offsets,
+		UBool  	flush,
+		UErrorCode *  	err 
+	)
+*/
```

Comment:
Whats purpose of that commented out code?

To delete?

Reply:
it's documentation.